File: C/Exercice4_4_fils.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main()
{
    int tube, retour;
    char buffer[256],
         *tube_str = getenv("TUBE_FD");

    if(tube_str == NULL)
    {
        puts("Erreur lors de la recuperation de la variable d'environnement\n");
        retour = 1;
    }
    else
    {
        tube = atoi(tube_str);
        read(tube, buffer, sizeof(buffer));

        printf("Le message du pere: %s\n", buffer);
        close(tube);
        retour = 0;
    }

    return retour;
}

File: Python/Exercice4_4_fils.py
#coding: utf-8
import os
import sys

tube_str = os.environ.get("TUBE_FD")
if tube_str is None:
    print("Erreur lors de la recuperation de la variable d'environnement\n")
    sys.exit(1)

tube = int(tube_str)
buffer = os.read(tube, 256).decode()
print(f"Le message du pere: {buffer}")
os.close(tube)

sys.exit(0)